lib/AST/ASTSearchTask.cpp

```diff
@@ -0,0 +1,44 @@
+#include "mull/AST/ASTSearchTask.h"
+
+#include "mull/AST/ASTVisitor.h"
+#include "mull/Parallelization/Progress.h"
+
+using namespace mull;
+
+ASTSearchTask::ASTSearchTask(mull::Diagnostics &diagnostics, mull::ASTStorage &astStorage,
+                             mull::MutatorKindSet mutatorKindSet, mull::FilePathFilter &pathFilter)
+    : diagnostics(diagnostics), astStorage(astStorage), mutatorKindSet(mutatorKindSet),
+      pathFilter(pathFilter) {}
+
+void ASTSearchTask::operator()(iterator begin, iterator end, Out &storage,
+                               mull::progress_counter &counter) {
+
+  std::vector<std::pair<SourceFilePath, SingleASTUnitMutations>> mutationUnits;
+
+  for (auto it = begin; it != end; it++, counter.increment()) {
+    const std::unique_ptr<mull::Bitcode> &bitcode = *it;
+
+    const llvm::Module *const module = bitcode->getModule();
+    std::string sourceFileName = module->getSourceFileName();
+    if (pathFilter.shouldSkip(sourceFileName)) {
+      diagnostics.info(std::string("ASTSearchTask: skipping path: ") + sourceFileName);
```

Comment:
Are you sure you want to put it into the "main" communication channel?

Reply:
Of course not! Thanks!